dora/core/client/fs/src/main/java/alluxio/client/file/dora/ConsistentHashProvider.java

```diff
@@ -0,0 +1,219 @@
+package alluxio.client.file.dora;
+
+import alluxio.Constants;
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.wire.WorkerNetAddress;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.hash.HashFunction;
+
+import javax.annotation.Nullable;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.stream.Collectors;
+
+import static com.google.common.hash.Hashing.murmur3_32_fixed;
+import static java.lang.Math.ceil;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+@VisibleForTesting
+public class ConsistentHashProvider {
+  private static final HashFunction HASH_FUNCTION = murmur3_32_fixed();
```

Comment:
this hash function is marked obsolete, we should replace it in a separate PR

Reply:
good call. I think this belongs to the microbenchmark field and we should probably do some tests comparing the hash options.